Jenkinsfile

```diff
@@ -40,13 +41,50 @@ def allArchitectures = [
 ]
 jobProperties.add(parameters([text(defaultValue: allArchitectures.join('\n'),
         description: 'The architectures (cheribuild suffixes) to build for (one per line)',
-        name: 'architectures')]))
+        name: 'architectures'),
+        text(defaultValue: ["riscv64-hybrid", "riscv64-purecap"].join('\n'),
+        description: 'The architectures (cheribuild suffixes) to build a purecap kernel for (one per line)',
+        name: 'purecapKernelArchitectures')]))
 // Set the default job properties (work around properties() not being additive but replacing)
 setDefaultJobProperties(jobProperties)
 
 jobs = [:]
 
-def buildImageAndRunTests(params, String suffix) {
+GlobalVars.selectedPurecapKernelArchitectures = params.purecapKernelArchitectures.split('\n')
+
+def makeTestStep(params, String suffix, testExtraArgs, extraArgs) {
+    return {
+        {
```

Comment:
```suggestion
```
You probably don't need this extra closure?

Reply:
yeah I don't think so, will remove after CI is done.